src/main/webapp/app/entities/quiz-exercise/quiz-exercise-detail.directive.ts

```diff
@@ -176,6 +192,99 @@ class QuizExerciseDetailController {
         }]);
     }
 
+    /**
+     * Toggles existing questions view
+     */
+    showHideExistingQuestions() {
+        if (typeof this.quizExercise === 'undefined') {
+            this.quizExercise = this.entity;
+        }
+
+        if (this.courses.length === 0) {
+            this.courseRepository.query().subscribe(
+                (res: HttpResponse<Course[]>) => {
+                    this.courses = res.body;
+                }
+            );
+        }
+        this.showExistingQuestions = !this.showExistingQuestions;
+    }
+
+    /**
+     * Populates list of quiz exercises for the selected course
+     */
+    onCourseSelect() {
+        this.allExistingQuestions = [];
+        if (this.selectedCourse === null) {
+            return;
+        }
+        const course = JSON.parse(this.selectedCourse);
+        this.repository.findForCourse(course.id)
+            .subscribe((quizExercisesResponse: HttpResponse<QuizExercise[]>) => {
+                if (quizExercisesResponse.body) {
+                    const quizExercises = quizExercisesResponse.body;
+                    for (const quizExercise of quizExercises) {
+                        this.repository.find(quizExercise.id).subscribe((response: HttpResponse<QuizExercise>) => {
+                            const quizExerciseResponse = response.body;
+                            for (const question of quizExerciseResponse.questions) {
+                                question.exercise = quizExercise;
+                                this.allExistingQuestions.push(question);
+                            }
+                            this.applyFilter();
+                        });
+                    }
+                } else {
+                    this.onSaveError();
+                }
+            });
+    }
+
```

Comment:
please remove the following outcommented code

Reply:
done,